### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Description
This PR adds several improvements by:
- Adding our usual editor config
- Adding linting for test files
- Adding prettier (and appliying it)
- Adding prettier to eslint
- Removing the license information from the PR template file (it's annoying to have this in github when opening a PR)
- Adjusting the workflow files to other projects
- Adding Jetbrains Copyright configuration
- Pinning dev dependencies
- Adding yarn berry
- Adding a yarn.lock so renovate can work

### Steps

- [x] Added changes
- [x] I read the [contribution documentation](https://github.com/hedgedoc/html-to-react/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: .github/workflows/reuse.yml [hunk 1]
```diff
@@ -1,10 +1,14 @@
-# SPDX-FileCopyrightText: 2020 The HedgeDoc developers (see AUTHORS file)
+# SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .github/workflows/reuse.yml:1]:
```suggestion
# SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .prettierignore.license [hunk 2]
```diff
@@ -0,0 +1,4 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .prettierignore.license:1]: 2022

File: .prettierrc.json.license [hunk 3]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .prettierrc.json.license:1]: 2022

File: .reuse/dep5 [hunk 4]
```diff
@@ -3,3 +3,14 @@ Upstream-Name: html-to-react
 Upstream-Contact: The HedgeDoc developers <license@hedgedoc.org>
 Source: https://github.com/hedgedoc/html-to-react
 
+Files: .github/pull_request_template.md
+Copyright: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .reuse/dep5:7]: 2022

File: .reuse/dep5 [hunk 5]
```diff
@@ -3,3 +3,14 @@ Upstream-Name: html-to-react
...
+License: CC-BY-SA-4.0
+
+Files: .yarn/**
+Copyright: Yarn contributors
+License: LicenseRef-yarn
+
+Files: .idea/**
+Copyright: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .reuse/dep5:15]: 2022

File: package.json [hunk 6]
```diff
@@ -6,11 +6,10 @@
...
     "test": "jest",
-    "build": "tsc",
-    "prepublishOnly": "yarn format && yarn test && yarn build",
-    "lint": "eslint . --ext .ts",
-    "format": "prettier -c \"{src,test}/**/*.{ts,tsx,js}\"",
-    "format:fix": "prettier -w \"{src,test}/**/*.{ts,tsx,js}\""
+    "build": "tsc -p tsconfig.build.json",
+    "prepublish": "rm -rf dist && yarn lint && yarn build && yarn test",
+    "lint": "eslint src --ext .ts",
+    "lint:fix": "eslint --fix --ext .ts src"
```
Comment (DerMolly) [inline package.json:12]: No more `format`?

File: tsconfig.build.json.license [hunk 7]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline tsconfig.build.json.license:1]:
```suggestion
SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: package.json (hunk 6 above)
Comment (mrdrogdrog) [reply package.json:12]: I added the eslint plugin for prettier. Therefore lint checks the code format using prettier and lint:fix reformats the code.

**Review by DerMolly — APPROVED** [review]

**Pull request merged**